src/python/gudhi/representations/vector_methods.py

```diff
@@ -10,6 +10,7 @@
 #   - 2021/11 Vincent Rouvreau: factorize _automatic_sample_range
 
 import numpy as np
+from scipy.spatial.distance import cdist
```

Comment:
You say "cdists handles inf/nan better than sklearn" in the commit. Could you
- add a comment in the code saying the same thing (so we are not tempted to change back later, or at least we know what to watch out for)
- explain here what the issue was? (mostly out of curiosity)

Reply:
In the Atol transform method I would rather this pairwise distance transmit/transform information (what cdist does, see below) than raise a ValueError (what sklearn.metrics.pairwise assert does).
`
import numpy as np
from scipy.spatial.distance import cdist
from sklearn.metrics import pairwise
measure = np.array([[0, np.inf], [0, 1], [0, np.nan])
centers = np.array([[0, 1], [0, 0]])
np.exp(-pairwise.pairwise_distances(X=measure, Y=centers))
np.exp(-cdist(XA=measure, XB=centers))
`
I am happy to have cdist compute exp(-infinity) (cdist) instead of 
> ValueError: Input contains infinity
from sklearn. What do you think?

Reply:
cdist computes the distance naively, which may involve infinity or NaN, whereas sklearn asserts that there is no infinity or NaN at any point in the computation (they have an option `force_all_finite=False`, but it seems to be limited to some metrics). Hmm, yes, accepting infinities seems ok.

So, can you add a comment in the code?